libloki/proof-of-work.js

```diff
@@ -0,0 +1,70 @@
+const hash = require('js-sha512');
+const bb = require('bytebuffer');
+
+// Increment Uint8Array by 1 with carrying
+function incrementNonce(nonce) {
+  idx = nonce.length - 1;
+  nonce[idx] += 1;
+  // Nonce will just reset to 0 if all values are 255 causing infinite loop, should never happen
+  while (nonce[idx] == 0 && idx >= 0) {
+    nonce[--idx] += 1;
+  }
+  return nonce;
+}
+
+// Convert a Uint8Array to a base64 string. Copied from stackoverflow
+function bufferToBase64(buf) {
+  var binstr = Array.prototype.map.call(buf, function (ch) {
+      return String.fromCharCode(ch);
+  }).join('');
+  return bb.btoa(binstr);
+}
+
+// Convert number to Uint8Array
+function numberToUintArr(numberVal) {
+  // TODO: Make this not hardcoded for arrays of length 8?
+  var arr = new Uint8Array(8);
+  for (var idx = 7; idx >= 0; idx--) {
+    // Get the lowest 8 bits from the number
+    var byte = numberVal & 0xff;
+    arr[idx] = byte;
+    // Essentially bitshift 
+    numberVal = (numberVal - byte) / 256 ;
```

Comment:
I guess actually using a bitshift would be more explicit, and the comment can then go !
```suggestion
    numberVal = (numberVal - byte) >> 8 ;
```

Reply:
As per offline discussion, bitwise operators are no-nos in js, so ignore this comment.